federation/cmd/federated-apiserver/app/server.go

```diff
@@ -390,9 +240,8 @@ func Run(s *options.APIServer) error {
 
 	var serviceAccountGetter serviceaccount.ServiceAccountTokenGetter
 	if s.ServiceAccountLookup {
-		// If we need to look up service accounts and tokens,
-		// go directly to etcd to avoid recursive auth insanity
-		serviceAccountGetter = serviceaccountcontroller.NewGetterFromStorageInterface(etcdStorage)
+		// TODO: Maybe do not expose this option in genericapiserver, if most servers do not need it?
```

Comment:
(pardon my ignorance) are service account tokens differently formatted from other tokens? How do we know that they are service account tokens and not some other sort? Do we not need to check other tokens for revocation?